File: java/game/src/main/java/example/Player.java
package example;

// This class showcases method overriding in Java.
// It extends the Character class and provides a specific implementation for the attack method.
public class Player extends Character {
    public Player(String name, int health) {
        super(name, health);
    }

    @Override
    public void attack() {
        System.out.println(name + " attacks with sword!");
    }

}

File: csharp/Game/Player.cs
namespace Game{

    public class Player(string name, int health) : Character(name, health){

        // In C#, the 'override' keyword is used to indicate that a method overrides a base class method.
        // This is similar to the @Override annotation in Java.
        public override void Attack(){
            Console.WriteLine($"{Name} attacks with a sword!");
        }
        
    }
}